airflow/configuration.py

```diff
@@ -473,13 +473,7 @@ def get_sections_including_defaults(self) -> list[str]:
 
         :return: list of section names
         """
-        my_own_sections = self.sections()
-
-        all_sections_from_defaults = list(self.configuration_description.keys())
-        for section in my_own_sections:
-            if section not in all_sections_from_defaults:
-                all_sections_from_defaults.append(section)
-        return all_sections_from_defaults
+        return list({**dict.fromkeys(self.configuration_description), **dict.fromkeys(self.sections())})
```

Comment:
I wonder if this should be further improved to

```python
import itertools

list(dict.fromkeys(itertools.chain(self.configuration_description, self.sections())))
```

but maybe this is too minor.